spec/jobs/topic_timer_enqueuer_spec.rb

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Jobs::TopicTimerEnqueuer do
+  subject { described_class.new }
+
+  fab!(:timer1) do
+    Fabricate(:topic_timer, execute_at: 1.minute.ago, created_at: 1.hour.ago, status_type: TopicTimer.types[:close])
+  end
+  fab!(:timer2) do
+    Fabricate(:topic_timer, execute_at: 1.minute.ago, created_at: 1.hour.ago, status_type: TopicTimer.types[:open])
+  end
+  fab!(:future_timer) do
+    Fabricate(:topic_timer, execute_at: 1.hours.from_now, created_at: 1.hour.ago, status_type: TopicTimer.types[:close])
+  end
+  fab!(:deleted_timer) do
+    Fabricate(:topic_timer, execute_at: 1.minute.ago, created_at: 1.hour.ago, status_type: TopicTimer.types[:close])
+  end
+
+  before do
+    deleted_timer.trash!
+  end
```

Comment:
```suggestion
  fab!(:deleted_timer) do
    Fabricate(:topic_timer, execute_at: 1.minute.ago, created_at: 1.hour.ago, status_type: TopicTimer.types[:close]).tap(&:trash!)
  end
```

I think this will work?

Reply:
Does not work, it cannot find the timer inside the it block later. IMO the current way is clearer anyway